### PR TITLE
8326964: Remove Eclipse Shared Workspaces

### PR DESCRIPTION
**Pull request by TheShermanTanker**
Eclipse Shared Workspaces share the same directory as the JDK itself, which does not make much sense since there can be multiple different configurations of the JDK which Eclipse does depend on (as minimalistic as it is), and there is no one true JDK configuration, unlike what this setting suggests. This feature was created when I was much less familiar with the structure of how the JDK handles different builds, and should be removed

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8326964](https://bugs.openjdk.org/browse/JDK-8326964): Remove Eclipse Shared Workspaces (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18046/head:pull/18046` \
`$ git checkout pull/18046`

Update a local copy of the PR: \
`$ git checkout pull/18046` \
`$ git pull https://git.openjdk.org/jdk.git pull/18046/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18046`

View PR using the GUI difftool: \
`$ git pr show -t 18046`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18046.diff">https://git.openjdk.org/jdk/pull/18046.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18046#issuecomment-1969195953)

### CODE REVIEW

**Comment by TheShermanTanker** [comment]
This should be ready for review now

**Review by erikj79 — APPROVED** [review]

**Comment by TheShermanTanker** [comment]
@erikj79 Sorry it seems I forgot to remove the SHARED definition from the Makefile, could I get a re\-review? Thanks
@magicus Also waiting for your approval

File: .gitignore
Comment (magicus) [inline .gitignore:22]: `.project` is repeated so that is fine to remove, but why are you removing the other two? `.cproject` sounds like it could come from CDT.

Comment (TheShermanTanker) [reply .gitignore:22]: With the Shared Workspaces removed, the Workspace created no longer exists in the JDK's top level source directory. This means the .gitignore entries are now obsolete, and can be removed, since they only ignore .cproject and .classpath in the top level source directory. They were added way back when I introduced support for Eclipse, and was less familiar with how multiple different JDK builds are contained

Comment (TheShermanTanker) [reply .gitignore:22]: Side tangent: How are the Workspaces for the other IDEs set up? Are they contained entirely in the ide subdirectory in the build directory or do they end up leaving their metadata in the top level directory like the Shared Workspaces used to do? I plan to improve Eclipse support over time, and these details might potentially be important to that effort

Comment (magicus) [reply .gitignore:22]: Ok, it's fine for integration then.

As for other IDEs: there is very little coordination on this; most IDE integrations are like barely kept alive by some enthusiast. I would really love to see some better grip on this, unifying IDE support and making it more of a first-class citizen of the build, but this has unfortunately kept being pushed down the prio list. :-(

If you feel so inclined, please try out the different IDEs supported. The very first step, I believe, is to have proper documentation in docs/ide.md. Even if we have a process that requires several manual steps, it is better to have it formalized and documented in a single place. Then it can form the basis for a more automated approach.

Comment (TheShermanTanker) [reply .gitignore:22]: Sorry I think I may not understand what you mean by unifying IDE support "and making it more of a first class citizen of the build", do you mean a central Setup IDE function in make of some sort, much like SetupNativeCompilation? I don't see how that might be possible given how each IDE has wildly varying requirements and save data formats

Comment (magicus) [reply .gitignore:22]: I think the most confusing part is how different all IDE setup procedures are. The unification I'd like to see is rather like `make ide-setup IDE=vscode AREA=hotspot,jdk` or something like that. That is, a clear and easy to understand way for the user to call a make target to create an environment for their IDE of choice. I realize that the actual implementation will be very varying for different IDEs, but I also think many share similar parts (like creating e.g. an xml file describing the project), and that there certainly are room for even some unification in that part.

**Comment by TheShermanTanker** [comment]
Thanks Erik and Magnus for the reviews

**Pull request closed without merging**
